tests/aws/services/lambda_/test_lambda_integration_sqs.py

```diff
@@ -1168,5 +1172,119 @@ def test_sqs_invalid_event_filter(
+        mapping_uuid = None
+
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=TEST_LAMBDA_PYTHON_ECHO_VERSION_ENV,
+            runtime=Runtime.python3_11,
+            role=lambda_su_role,
+        )
+
+        aws_client.lambda_.update_function_configuration(
+            FunctionName=function_name, Environment={"Variables": {"CUSTOM_VAR": "a"}}
+        )
+        aws_client.lambda_.get_waiter("function_updated_v2").wait(FunctionName=function_name)
+        publish_v1 = aws_client.lambda_.publish_version(FunctionName=function_name)
+        aws_client.lambda_.get_waiter("function_active_v2").wait(
+            FunctionName=publish_v1["FunctionArn"]
+        )
 
-# TODO: test integration with lambda logs
+        queue_url_1 = sqs_create_queue(QueueName=queue_name_1)
+        queue_arn_1 = sqs_get_queue_arn(queue_url_1)
+        create_event_source_mapping_response = aws_client.lambda_.create_event_source_mapping(
+            EventSourceArn=queue_arn_1,
+            FunctionName=publish_v1["FunctionArn"],
+            MaximumBatchingWindowInSeconds=1,
+        )
+        mapping_uuid = create_event_source_mapping_response["UUID"]
+        cleanups.append(lambda: aws_client.lambda_.delete_event_source_mapping(UUID=mapping_uuid))
+        snapshot.match("create-event-source-mapping-response", create_event_source_mapping_response)
+        _await_event_source_mapping_enabled(aws_client.lambda_, mapping_uuid)
+
+        aws_client.sqs.send_message(QueueUrl=queue_url_1, MessageBody=json.dumps({"foo": "bar"}))
+
+        events = retry(
+            check_expected_lambda_log_events_length,
+            retries=10,
+            sleep=1,
+            function_name=function_name,
+            expected_length=1,
+            logs_client=aws_client.logs,
+        )
+        snapshot.match("events", events)
+
+        rs = aws_client.sqs.receive_message(QueueUrl=queue_url_1)
+        assert rs.get("Messages") is None
+
+        # # create new function version
+        aws_client.lambda_.update_function_configuration(
+            FunctionName=function_name, Environment={"Variables": {"CUSTOM_VAR": "b"}}
+        )
+        aws_client.lambda_.get_waiter("function_updated_v2").wait(FunctionName=function_name)
+        publish_v2 = aws_client.lambda_.publish_version(FunctionName=function_name)
+        aws_client.lambda_.get_waiter("function_active_v2").wait(
+            FunctionName=publish_v2["FunctionArn"]
+        )
+        # we're now pointing the existing event source mapping towards the new version.
+        # only v2 should now be called
+        updated_esm = aws_client.lambda_.update_event_source_mapping(
+            UUID=mapping_uuid, FunctionName=publish_v2["FunctionArn"]
+        )
+        assert mapping_uuid == updated_esm["UUID"]
+        assert publish_v2["FunctionArn"] == updated_esm["FunctionArn"]
+        # snapshot.match("updated_esm", updated_esm)
```

Comment:
Is this commented out intentionally?

Reply:
I didn't know, so I un-commented and re-ran the test to create the snapshot and it seems to be working.